src/app/shared/components/cities-search/display-location-option.ts

```diff
@@ -20,8 +20,11 @@ export function displayLocationOption(location?: Location) {
   return toTitleCase(location.city) + ', ' + location.region;
 
   function toTitleCase(value: string) {
-    // DO NOT replace it with `value[0].toUpperCase()` etc. as that may not
-    // select the whole glyph, unlike unicode regex replacement.
-    return value.toLowerCase().replace(/^./u, (letter) => letter.toUpperCase());
+    // ^. - first character of the string
+    // \s. - first character after whitespace
+    // -. - first character after - (f.e. Kudowa-Zdroj)
+    return value
+      .toLowerCase()
+      .replace(/(^.|\s.|-.)([A-Za-zĄąĆćĘęŁłŃńÓóŚśŹźŻż]*)/g, (_, m1, m2) => m1.toUpperCase() + m2.toLowerCase());
```

Comment:
This won't scale to more languages. Use a proper Unicode Property Escape instead. Also, no need to call `toLowerCase()` on anything since you're doing that in the preceding line for the whole value:
```suggestion
      .replace(/(^.|\s.|-.)(\p{Letter}*)/g, (_, m1, m2) => m1.toUpperCase() + m2);
```

Reply:
Good point! Regular expression has been improved to more flexible one. Thanks for your vigilance!